sympy/simplify/simplify.py

```diff
@@ -614,6 +615,10 @@ def shorter(*choices):
     if expr.has(Product):
         expr = product_simplify(expr)
 
+    if expr.has(KroneckerDelta):
+        from sympy.functions.elementary.piecewise import Piecewise
+        expr = simplify(expr.rewrite(Piecewise))
```

Comment:
Is the call to simplify needed here? Maybe this could just be piecewise_fold instead of simplify. Or since piecewise_fold is called above maybe the rewrite could be moved above that...

Reply:
Also it looks as if it would be okay to import Piecewise at the top.